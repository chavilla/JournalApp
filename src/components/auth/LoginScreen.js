import React from "react";
import { useDispatch } from 'react-redux'
import { Link } from "react-router-dom";
import { login } from "../../actions/auth";
import useForm from "../../hooks/useForm";

const LoginScreen = ( ) => {

  const dispatch = useDispatch();

  const [value, handleChange]=useForm({
    email: '',
    password:'',
  });

  const  { email, password }=value;

  const handleLogin=(e)=>{
    e.preventDefault();
    dispatch(login('d45y78', 'Jesús'));
  }

  return (
    <>
      <h3 className='auth__title'>Iniciar Sesión</h3>
      <form
      onSubmit={ handleLogin }
      >
        <div className='auth__container-input'>
        <input className='auth__input' 
        type="email" name="email" autoComplete='off' placeholder='Email' value={email} 
        onChange={handleChange}
        />
        </div>

        <div className='auth__container-input'>
        <input className='auth__input' type="password" name="password" placeholder='Password' value={password} 
        onChange={handleChange}
        />
       
        </div>

        <button className='btn btn-primary btn-block mt-1' type="submit">Login</button>
        
        <div className='auth__social-network'>
          <p>Login with Social Networks</p>
          <div className="google-btn">
            <div className="google-icon-wrapper">
              <img
                className="google-icon"
                src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
                alt="google button"
              />
            </div>
            <p className="btn-text">
              <b>Sign in with google</b>
            </p>
          </div>
        </div>
        <Link to='/auth/register' className='link'>
            Create an account
        </Link>
      </form>
    </>
  );
};

export default LoginScreen;
