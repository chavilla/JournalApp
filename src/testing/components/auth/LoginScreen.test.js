import { mount } from "enzyme";
import React from "react";
import { Provider } from "react-redux";
import LoginScreen from "../../../components/auth/LoginScreen";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import { initState2 } from "../../../fixtures/";
import { MemoryRouter } from "react-router-dom";

// settings
const midlewares = [thunk];
const mockStore = configureStore(midlewares);
let store = mockStore(initState2);

const history = {
    listen: jest.fn(),
}

describe("Testing in LoginScreen", () => {
  beforeEach(() => {
    store = mockStore(initState2);
  });

  test("should to create a snapshot", () => {
    const wrapper = mount(
      <Provider store={store}>
          <MemoryRouter history={ history}>
            <LoginScreen />
          </MemoryRouter>
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();

  });
});
