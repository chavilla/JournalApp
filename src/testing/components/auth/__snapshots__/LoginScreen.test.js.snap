// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in LoginScreen should to create a snapshot 1`] = `
Array [
  <h3
    className="auth__title"
  >
    Iniciar Sesi√≥n
  </h3>,
  <form
    className="animate__animated animate__fadeIn"
    onSubmit={[Function]}
  >
    <div
      className="auth__container-input"
    >
      <input
        autoComplete="off"
        className="auth__input"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="email"
        value=""
      />
      <div
        className="auth__alert-error"
      />
    </div>
    <div
      className="auth__container-input"
    >
      <input
        className="auth__input"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <div
        className="auth__alert-error"
      />
    </div>
    <button
      className="btn btn-primary btn-block mt-1"
      disabled={false}
      type="submit"
    >
      Login
    </button>
    <div
      className="auth__social-network"
    >
      <p>
        Login with Social Networks
      </p>
      <div
        className="google-btn"
        onClick={[Function]}
      >
        <div
          className="google-icon-wrapper"
        >
          <img
            alt="google button"
            className="google-icon"
            src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
          />
        </div>
        <p
          className="btn-text"
        >
          <b>
            Sign in with google
          </b>
        </p>
      </div>
    </div>
    <a
      className="link"
      href="/auth/register"
      onClick={[Function]}
    >
      Create an account
    </a>
  </form>,
]
`;
